game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -1441,6 +1431,22 @@ private static void placeUnits(final TriggerAttachment t, final Territory terr,
   public static void triggerNotifications(final Set<TriggerAttachment> satisfiedTriggers, final IDelegateBridge bridge,
       final String beforeOrAfter, final String stepName, final boolean useUses, final boolean testUses,
       final boolean testChance, final boolean testWhen) {
+
+    // NOTE: The check is needed to ensure that UI-related code (namely 'NotificationMessages.getInstance()') is
+    // not executed when unit testing non-UI related code such as 'MustFightBattleTest'.
+    if (satisfiedTriggers.stream().anyMatch(notificationMatch())) {
```

Comment:
Thanks for digging into this, looks like we're still a long way from a good modular design.
However the overloaded method is a good way to add unit tests nevertheless. 👍